src/Extensions/Abstractions/ReplayableLogs/TimedScopeDefinition.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+
+namespace Microsoft.Omex.Extensions.Abstractions
+{
+	/// <summary>
+	/// Store Timed Scope name
+	/// </summary>
+	public readonly struct TimedScopeDefinition
+	{
+		/// <summary>
+		/// Timed Scope name
+		/// </summary>
+		public string Name { get; }
+
+
+		/// <summary>
+		/// TimedScopeDefinition constructor
+		/// </summary>
+		/// <param name="name">Timed Scope name</param>
+		/// <exception cref="ArgumentException">Thrown when name null or empty</exception>
+		public TimedScopeDefinition(string name) =>
+			Name = string.IsNullOrWhiteSpace(name)
```

Comment:
I was thinking about replacing it with `Debug.Assert`, but it looks too risky to me since most of our automated builds using `Release` configuration, and const of this check not that heavy to losing a luxury of proper exception.
If we will change our integrations builds to build both configurations in the future we can revisit this topic.